server/src/test/java/org/elasticsearch/action/support/replication/ReplicationOperationTests.java

```diff
@@ -439,10 +440,15 @@ public void testWaitForActiveShards() throws Exception {
         final int unassignedReplicas = randomInt(2);
         final int totalShards = 1 + assignedReplicas + unassignedReplicas;
         final int activeShardCount = randomIntBetween(0, totalShards);
+        final boolean stateless = randomBoolean();
         Request request = new Request(shardId).waitForActiveShards(
             activeShardCount == totalShards ? ActiveShardCount.ALL : ActiveShardCount.from(activeShardCount)
         );
-        final boolean passesActiveShardCheck = activeShardCount <= assignedReplicas + 1;
+        // In the case of stateless, only the search/replica assigned shards are calculated as active shards. But in other cases, or when
```

Comment:
Maybe not mention stateless, but instead "unpromotable" (which has a meaning in core)